src/rules/eva-voice-biometric.js

```diff
@@ -0,0 +1,143 @@
+/**
+*	@title EVA Voice Biometric connector
+*	@overview EVA Voice Biometric connector rule for Auth0 enables voice enrolment and verification as a second factor
+*	@gallery true
+*	@category marketplace
+*	@author Auraya Systems 2020
+*
+*	EVA WEB is a second-factor voice biometric solution designed for use with a range of web browsers that support audio capture. 
+*	The biometric verification process is intended to be part of an authentication flow to enhance security, privacy and ease-of-use. 
+*	EVA WEB is an offering from Auraya Systems.
+*
+*	The voice tokens used with EVA WEB are predominantly digit strings, although phrases are also supported. 
+*	The usual verification process will usually prompt the user to read out a sequence of random digits, 
+*	a sequence of consecutive digits, and optionally their phone number.
+*
+*	The first time a user is directed to EVA WEB where they will be invited to enrol their voice. 
+*	This process captures between 3 and 8 utterances depending on the required voice tokens, and includes a biometric “Opt-In” consent step. 
+*	At the conclusion of enrolment they will be redirected back to the authentication flow.
+*	
+*	Configuration Items:
+*	EVA_URL = EVA endpoint, typically: https://eva-web.mydomain.com/server/oauth
+*	CLIENT_ID = JWT client id on the EVA server (and this server)
+*	CLIENT_SECRET = JWT client secret on the EVA server (and this server)
+*	ISSUER = this app (or "issuer")
+*	
+*	RANDOM_DIGITS = true|false
+*	COMMON_DIGITS = true|false
+*	PERSONAL_DIGITS = a user.user_metadata field that contains digits such as phone_number, or blank
+*/
+function(user, context, callback) {
+
+	console.log(user);
+	//console.log(context);
+	console.log(configuration);
+
+	var clientSecret = configuration.CLIENT_SECRET;
+    
+  
+	// Prepare Users' enrolment status
+	user.user_metadata = user.user_metadata || {};
+	user.user_metadata.eva = user.user_metadata.eva || {};
+
+
+	// User has initiated a login and is prompted to use voice biometrics
+	// Send user's information and query params in a JWT to avoid tampering
+	function createToken(clientId, clientSecret, issuer, user) {
```

Comment:
The first 3 parameter values here are in the parent function's scope so no need to pass them in.

Reply:
ok